File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
import java.util.*;

class Solution {
    public String solution(String code) {
        StringBuilder sb = new StringBuilder();
        
        int mode = 0;
        for (int i=0; i<code.length(); i++) {
            char ch = code.charAt(i);
            
            if (ch == '1') {
                mode = mode == 1 ? 0 : 1;
                continue;
            }
            
            if (i % 2 == mode) {
                sb.append(ch);
            }
        }
        return sb.length() != 0 ? sb.toString() : "EMPTY";
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.kt
import java.util.*

class Solution {
    fun solution(code: String): String {
        val sb = StringBuilder()
        var mode = 0
        
        code.forEachIndexed { index, ch ->
            if(ch == '1') mode = mode xor 1
            else if (index % 2 == mode) {
                sb.append(ch);
            }
        }
        
        return if (sb.isNotEmpty()) {
            sb.toString()
        } else {
            "EMPTY"
        }
    }
}